src/Components/Components/src/ICascadingParameterAttribute.cs

```diff
@@ -0,0 +1,12 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.Components;
+
+// Represents an attribute marking a parameter to be set by a cascading value.
+// This exists so cascading parameter attributes can be defined outside the Components assembly.
+// For example: [SupplyParameterFromForm].
+internal interface ICascadingParameterAttribute
+{
+    public string? Name { get; set; }
+}
```

Comment:
I would rather change `CascadingParameterAttribute` to be nonsealed and thereby not require a marker interface. The forms and query ones can then just inherit it.

This does mean that developers can make their own subclasses but I don't see any reason to object to that. Unless we make `ICascadingValueSupplier` public, subclassing `CascadingParameterAttribute` would not allow people to do anything different from just using `CascadingParameterAttribute` as they already do today.

Reply:
> This does mean that developers can make their own subclasses but I don't see any reason to object to that.

There's no benefit to it either, is it?

The consequences of doing it that way is that we have created an "infinite set" of attributes that can mean "I am a cascading value", and that's something we'll have to account for in the future.

For example, if we do source generation based on the attribute, now we need to pay the cost of performing semantic analysis over all the parameters, as we need to inspect each attribute and their base class for determining its a target, instead of limiting the scope to the concrete set of attributes we care about.

Reply:
The benefit is just that this relationship corresponds naturally to inheritance, as we want to share code across these types anyway. Is there a problem with using Roslyn's semantic model to tell us if a type inherits from another? Seems like that's exactly what it's designed for, and is no different from hundreds of other places in the codebase that allow subclassing.

It's not that I'm hugely fixated on using a subclass here, but I am getting concerned that our codebase is getting increasingly niche.

Reply:
The specific example here is making a nonobvious design choice that introduces extra concepts (marker interfaces), with a rationale based on a future possibility of optimizing source generation.

Reply:
That it's just a practical sample.

The rationale here is opening up a potential extensibility point without a clear benefit to it. Marker interfaces are a well-known concept that is used across our codebase.

The principle in contrast here is minimizing API surface and extensibility points unless there are clear scenarios that users can exercise.